src/sys/uio.rs

```diff
@@ -55,7 +55,7 @@ pub fn readv(fd: RawFd, iov: &mut [IoVec<&mut [u8]>]) -> Result<usize> {
 pub fn pwrite(fd: RawFd, buf: &[u8], offset: i64) -> Result<usize> {
     let res = unsafe {
         ffi::pwrite(fd, buf.as_ptr() as *const c_void, buf.len() as size_t,
-                    offset)
+                    offset as off_t)
```

Comment:
Would it not be better to change the type signature for `pread`/`pwrite` to take offset as an `off_t` rather than casting here?